src/Unread.ts

```diff
@@ -48,7 +48,7 @@ export function eventTriggersUnreadCount(ev: MatrixEvent): boolean {
     }
 
     if (ev.isRedacted()) return false;
-    return haveRendererForEvent(ev);
+    return haveRendererForEvent(ev, false /* hidden messages should never trigger unread counts anyways */);
```

Comment:
this is one of the few places we do need the implicit `true` behaviour: if we can't render the tile, it shouldn't be counted as unread.

This ties back to push rules being awful.

Reply:
That's what I was thinking too, but isn't it actually the implicit `false` behavior that we want? `true` would cause this to return a renderer for hidden events, counting them as unread.

Reply:
err, yes.... good call. This area makes my head hurt.